pkg/api/image/schema.go

```diff
@@ -20,15 +20,17 @@ func RegisterSchema(scaled *config.Scaled, server *server.Server, _ config.Optio
 	vmi := scaled.HarvesterFactory.Harvesterhci().V1beta1().VirtualMachineImage()
 	bids := scaled.LonghornFactory.Longhorn().V1beta2().BackingImageDataSource()
 	bi := scaled.LonghornFactory.Longhorn().V1beta2().BackingImage()
+	ctlcdi := scaled.CdiFactory.Cdi().V1beta1().DataVolume()
+	ctlcdiupload := scaled.CdiUploadFactory.Upload().V1beta1().UploadTokenRequest()
 
 	vmio := common.GetVMIOperator(vmi, vmi.Cache(), http.Client{})
 	downloaders := map[harvesterv1.VMIBackend]backend.Downloader{
 		harvesterv1.VMIBackendBackingImage: backingimage.GetDownloader(bi.Cache(), http.Client{}, vmio),
-		harvesterv1.VMIBackendCDI:          cdi.GetDownloader(),
+		harvesterv1.VMIBackendCDI:          cdi.GetDownloader(http.Client{}, scaled.Management.ClientSet, vmio),
 	}
 	uploaders := map[harvesterv1.VMIBackend]backend.Uploader{
 		harvesterv1.VMIBackendBackingImage: backingimage.GetUploader(bi.Cache(), bids, http.Client{}, vmio),
-		harvesterv1.VMIBackendCDI:          cdi.GetUploader(),
+		harvesterv1.VMIBackendCDI:          cdi.GetUploader(vmi, ctlcdi, ctlcdiupload, http.Client{}, vmio),
```

Comment:
Based on the original design of https://github.com/harvester/harvester/pull/7170, I'd like to have all backend implementations know about vmimage's details.

So ideally, this better to be:
```suggestion
		harvesterv1.VMIBackendCDI:          cdi.GetUploader(ctlcdi, ctlcdiupload, http.Client{}, vmio),
```

and the backend implementation operators vmimage via `vmio`.

However, considering our schedule, we may defer it to the discussion in #7524

Reply:
Updated.

I would like to get the latest VM Image to ensure the progress update.
So, I add a new function to get the VM Image from vmi operator.